src/natcap/invest/utils.py

```diff
@@ -538,6 +541,38 @@ def build_lookup_from_csv(
     return lookup_dict 
 
 
+def read_csv_to_dataframe(path, to_lower=False, encoding=None, **kwargs):
+    """
+    Wrapper around ``pandas.read_csv`` that standardizes the column names by
+    stripping leading/trailing whitespace and optionally making all lowercase.
+    This helps avoid common errors caused by user-supplied CSV files with
+    column names that don't exactly match the specification.
+
+    Args:
+        path (string): path to a CSV file
+        to_lower (bool): if True, convert all column names to lowercase
+        encoding (string): name of encoding codec to pass to `read_csv`. 
+            Defaults to None.
+        **kwargs: any kwargs that are valid for ``pandas.read_csv``
+    """
+    # Check if the file encoding is UTF-8 BOM first
+    # allow encoding kwarg to override this if it's provided
+    if not encoding:
+        with open(path, 'rb') as file_obj:
+            first_line = file_obj.readline()
+            if first_line.startswith(codecs.BOM_UTF8):
+                encoding = 'utf-8-sig'
+    dataframe = pandas.read_csv(path, encoding=encoding, **kwargs)
```

Comment:
Hey @emlys sorry, I thought I had read a comment somewhere that this logic could be entirely removed if we just defaulted to using `encoding='utf-8-sig'`, which is also what the python docs make it sound like is the case.  Is that not actually the case?